src/table/utils/styling-utils.ts

```diff
@@ -42,34 +27,68 @@ export function getColor(defaultColor: string, theme: stardust.Theme, color = {}
 /**
  * Gets font color based on background, making sure contrast is good enough
  */
-export const getAutoFontColor = (backgroundColor: string): string =>
-  isDarkColor(backgroundColor) ? StylingDefaults.WHITE : StylingDefaults.FONT_COLOR;
+export const getAutoFontColor = (background: string): string =>
+  isDarkColor(background) ? StylingDefaults.WHITE : StylingDefaults.FONT_COLOR;
 
 /**
  * get the border color based on the color of the background, returns bright color if background is dark and vice versa
+ * the borderLeftColor is only used for the leftmost cells, but that is added/removed by changing borderWidth
+ * the same thing applies to the borderTopColor which is only used for totals when it is at the bottom
  */
-export const getBorderColor = (isBackgroundDark: boolean): string => (isBackgroundDark ? '#F2F2F2' : '#D9D9D9');
+export const getBorderColors = (isBackgroundDark: boolean, bottomSeparatingBorder: boolean) => {
+  // TODO: proper borders for dark background
+  if (isBackgroundDark) {
+    return {
+      borderBottomColor: '#F2F2F2',
+      borderRightColor: '#F2F2F2',
+      borderLeftColor: '#F2F2F2',
+      borderTopColor: '#F2F2F2',
+    };
+  }
+
+  return {
+    borderLeftColor: '#D9D9D9',
+    borderRightColor: '#D9D9D9',
+    borderTopColor: '#808080',
+    borderBottomColor: bottomSeparatingBorder ? '#808080' : '#EBEBEB',
+  };
+};
+
+/**
+ * Get border widths for body. adds a bottom border if the rendered rows height is estimated to be greater than the container height
+ */
+const getLastRowBottomBorder = (fontSize: string | undefined, rowsLength?: number, rootElement?: HTMLElement) => {
+  if (fontSize && rowsLength && rootElement) {
+    // add line height (133%), padding and borders
+    const rowHeight = (+fontSize.slice(0, -2) * 4) / 3 + 9;
```

Comment:
3 + 9 seems to be a magic number, is it about some padding values?

Reply:
well that's not order of operations work ;) I don't like what the formatter does here, it makes it unclear. This is what happens. pixels * 4/3 + 9. 4/3 is 133% but more exact, and the paddding is 4px top and bottom plus bottom border. I could clarify it in the comment

Reply:
Yes, Thanks for clarification, 
I was wondering more about numbers 3 and 9 itself but now it is super clear specially for 9 which is the combination of three things ✅

Reply:
actually I could move some brackets around and it is much clearer